modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -171,13 +161,12 @@ Mat Transpose(
         outputDims.emplace_back(input_shape_override[dim]);
 
     Mat output;
-    // TODO: ouptimize
-    MatShape tmp_perm;
-    tmp_perm.reserve(permutation.size());
-    for (int i = 0; i < permutation.size(); i++)
-        tmp_perm.emplace_back(static_cast<int>(permutation[i]));
+    MatShape order;
```

Comment:
`MatShape` is a `typedef std::vector<int> MatShape`. So this can be replaced to `MatShape order(permutation.begin(), permutation.end());`

Reply:
done